src/edu/umass/ckc/wo/log/TeacherLogger.java

```diff
@@ -0,0 +1,137 @@
+package edu.umass.ckc.wo.log;
+
+import edu.umass.ckc.wo.db.DbProblem;
+import edu.umass.ckc.wo.event.NavigationEvent;
+import edu.umass.ckc.wo.smgr.SessionManager;
+import edu.umass.ckc.wo.event.tutorhut.*;
+import edu.umass.ckc.wo.tutor.response.HintResponse;
+import edu.umass.ckc.wo.tutor.response.ProblemResponse;
+import edu.umass.ckc.wo.tutor.response.Response;
+import edu.umass.ckc.wo.tutor.response.InterventionResponse;
+import edu.umass.ckc.wo.tutor.response.AttemptResponse;
+import edu.umass.ckc.wo.tutor.response.ExampleResponse;
+import edu.umass.ckc.wo.tutor.response.HintSequenceResponse;
+import edu.umass.ckc.wo.content.Hint;
+
+import java.sql.*;
+import java.util.List;
+
+/**
+
+ * Frank 	12-09-19	Issue #21 add teacher activity logger support
+
+ */
+public class TeacherLogger {
+    //private SessionManager smgr;
+    Connection conn;
+    public static final String TeacherEventLog = "teacherlog";
+
+    public static final String[] TEACHER_EVENTS = { "Login", "Logout" };
+
+    boolean tableExists = true;
+    
+
+
+    public TeacherLogger (Connection conn) {
+        //this.smgr = smgr;
+        this.conn = conn;
+        
+        testTable();
+        
+        if (!tableExists) {
+        	createTable();
+        }
+
+    }
+
+    private void testTable() {
+ 
+    	
+    	
+     String querytest = "SELECT COLUMN_NAME FROM information_schema.columns where table_name = 'teacherlog';"; 
+     try {
+
+    	    PreparedStatement ps = null;
+    	    ResultSet rs = null;        
+            ps = this.conn.prepareStatement(querytest);
+            rs = ps.executeQuery();
+            if (!rs.next()) {
+            	tableExists = false;
+        	}
+            ps.close();
+        }
+        catch (SQLException e) {
+            e.printStackTrace();
+        }        
+    }
+    
+    private void createTable() {

```

Comment:
let's remove the table creation out of it. we will Create the table separately.

Reply:
I removed the tablecreation